openedx/core/djangoapps/course_apps/handlers.py

```diff
@@ -0,0 +1,24 @@
+"""
+Signal handlers for course apps.
+"""
+from django.dispatch import receiver
+from openedx.core.djangoapps.course_apps.models import CourseAppStatus
+
+from openedx.core.djangoapps.course_apps.plugins import CourseAppsPluginManager
+from xmodule.modulestore.django import SignalHandler
+
+
+@receiver(SignalHandler.course_published)
+def update_course_apps(sender, course_key, **kwargs):  # pylint: disable=unused-argument
+    """
+    Whenever the course is published, update the status of course apps in the
+    django models to match their status in the course.
+    """
+    course_apps = CourseAppsPluginManager.get_apps_available_for_course(course_key)
+    for course_app in course_apps:
+        is_enabled = course_app.is_enabled(course_key=course_key)
+        CourseAppStatus.objects.update_or_create(
```

Comment:
This will be run on every course publish. Even if the course app status has not changed it will make the db call. Would it be better if we check if the status already exists and update only if it has been updated?

Reply:
We won't know if the course app status has changed until we actually check right? So we do need to run it on each course publish.